packit/utils/extensions.py

```diff
@@ -0,0 +1,82 @@
+# MIT License
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+packit.utils.extensions
+=======================
+
+Keeps functions that extend Python collections (list to dict, get on nested dict)
+and force strict evaluation (`assert_existence`).
+"""
+
+import json
+from typing import Any
+
+from packit.exceptions import PackitException
+
+
+def get_rev_list_kwargs(opt_list):
+    """
+    Converts the list of 'key=value' options to dict.
+    Options without value gets True as a value.
+    """
+    result = {}
+    for opt in opt_list:
+        opt_split = opt.split(sep="=", maxsplit=1)
+        if len(opt_split) == 1:
+            result[opt] = True
+        else:
+            key, raw_val = opt_split
+            try:
+                val = json.loads(raw_val.lower())
+                result[key] = val
+            except json.JSONDecodeError:
+                result[key] = raw_val
+    return result
+
+
+def assert_existence(obj):
```

Comment:
From what I see it's used only [in one place](https://github.com/packit/packit/blob/master/packit/api.py#L149).
So +1 to your suggestion to replace it with basic `assert` (which would tell us more than this generic error message).

Reply:
what about the optimized execution that ignores `assert`? I could probably enhance the error message in exception with the name of variable.

Reply:
I'm fine with both so if you're willing to enhance the message, go ahead.

Reply:
The `assert` should not be used in the regular code. It's meant only for tests* and can be skipped when using `-O` option for Python as @mfocko mentioned. I know this mode is not widely used, but we can still be consistent/safe in this. I couldn't find a better&cleaner solution for this. 

\* I can't find a good source/article for this now. Sorry.